### PR TITLE
feat(control): add plugins/reload to control api

### PR DESCRIPTION
**Pull request by sheharyaar**
### Description

`plugin/reload` is under admin API. The admin API can only be called at the control plane (if we set the role to control plane). But we need to call the plugin/reload API at the data plane, so if we set the role of the apisix to data plane we can't call this API.

This PR moves the `plugins/reload` to Control API so that we can call the endpoint from the data plane. For compatibility reasons, we also need to retain this API in admin API.

Refer: https://apisix.apache.org/docs/apisix/admin-api/#plugin

### Checklist

- [X] I have explained the need for this PR and the problem it solves
- [X] I have explained the changes or the new features added to this PR
- [X] I have added tests corresponding to this change
- [X] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/control/v1.lua [hunk 1]
```diff
@@ -400,6 +400,13 @@ function _M.dump_plugin_metadata()
     return 200, metadata.value
 end
 
+function _M.plugins_reload()
```
Comment (starsz) [inline apisix/control/v1.lua:403]: Missing the logic of sending events to the other worker processes.
You can refer: https://github.com/apache/apisix/blob/master/apisix/admin/init.lua#L284

**Comment by monkeyDluffy6017** [comment]
You mean `move`, do we need to remove the old API?

**Comment by sheharyaar** [comment]
@monkeyDluffy6017  @starsz please review

File: t/control/plugins-reload.t [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            local code, body, res = t.test('/v1/plugins_reload',
+                ngx.HTTP_POST)
+            ngx.say(res)
+        }
+    }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+done
```
Comment (starsz) [inline t/control/plugins-reload.t:52]: Can you add more test cases to test it ? 
You can refer https://github.com/apache/apisix/blob/master/t/admin/plugins-reload.t

**Comment by monkeyDluffy6017** [comment]
Do we need to remove the old API?

**Comment by monkeyDluffy6017** [comment]
Can you describe in detail why do you submit this pr?

File: apisix/control/router.lua [hunk 3]
```diff
@@ -46,6 +47,80 @@ local function format_dismod_uri(mod_name, uri)
...
+
+    local new_set = {}
+    for _, p in ipairs(new) do
+        new_set[p.name] = p
+    end
+
+    return core.table.set_eq(old_set, new_set)
+end
+
+local function sync_local_conf_to_etcd(reset)
```
Comment (starsz) [inline apisix/control/router.lua:64]: The code is the same as it is under admin API. Why not put it into the utils directory?

File: docs/en/latest/control-api.md [hunk 4]
```diff
@@ -475,3 +475,9 @@ Dumps the metadata with the specified `plugin_name`:
     "id": "file-logger"
 }
 ```
+
+### POST /v1/plugins_reload
```
Comment (starsz) [inline docs/en/latest/control-api.md:479]: The method is `PUT` right?

File: docs/en/latest/control-api.md [hunk 5]
```diff
@@ -475,3 +475,9 @@ Dumps the metadata with the specified `plugin_name`:
...
+
+Introduced in [v3.9.0](https://github.com/apache/apisix/releases/tag/3.9.0)
+
+Triggers a hot reload of the plugins.
```
Comment (starsz) [inline docs/en/latest/control-api.md:483]: Can you give an example of how to use this API?

File: apisix/control/v1.lua [hunk 6]
```diff
@@ -474,5 +485,12 @@ return {
         uris = {"/plugin_metadata/*"},
         handler = _M.dump_plugin_metadata,
     },
+    -- /v1/plugins_reload
+    {
+        methods = {"PUT"},
+        uris = {"/plugins_reload"},
```
Comment (starsz) [inline apisix/control/v1.lua:491]:
```suggestion
        uris = {"/plugins/reload"},
```
Would be better.

**Review by starsz — APPROVED** [review]

**Review by starsz — APPROVED** [review]

File: apisix/control/router.lua [hunk 7]
```diff
@@ -198,5 +201,26 @@ end
...
+
+    local local_conf = core.config.local_conf()
+    local deployment_role = core.table.try_read_attr(
+                       local_conf, "deployment", "role")
+    if deployment_role ~= "data_plane" then
+        -- data_plane should not write to etcd
+        if ngx_worker_id() == 0 then
+            sync_local_conf_to_etcd()
+        end
+    end
```
[the remark this answers is not in the document]
Comment (monkeyDluffy6017) [reply apisix/control/router.lua:216]: @starsz please help to check this

[the remark this answers is not in the document]
Comment (starsz) [reply apisix/control/router.lua:216]: The logic is the same as it in the admin API. 
https://github.com/apache/apisix/blob/master/apisix/admin/init.lua#L469-L475

File: apisix/utils/config_etcd.lua
[the remark this answers is not in the document]
Comment (monkeyDluffy6017) [reply apisix/utils/config_etcd.lua:1]: @sheharyaar

File: apisix/control/router.lua (hunk 7 above)
[the remark this answers is not in the document]
Comment (starsz) [reply apisix/control/router.lua:216]: After thinking about it, I think in the control API we don't need to update the `/plugins` resource in the etcd even if the role is `traditional`.

File: apisix/control/router.lua [hunk 8]
```diff
@@ -198,5 +197,4 @@ end
 
 end -- do
 
-
```
Comment (starsz) [inline apisix/control/router.lua:201]: Don't need to delete this line.

File: apisix/control/v1.lua [hunk 9]
```diff
@@ -400,6 +403,12 @@ function _M.dump_plugin_metadata()
     return 200, metadata.value
 end
 
+function _M.reload_plugins()
+    core.log.info("start to hot reload plugins")
+    plugin.load()
```
Comment (starsz) [inline apisix/control/v1.lua:408]: You should keep the logic of event.

**Review by starsz — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**